orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -155,11 +155,15 @@ def backward(ctx, *grad_outputs):
 
                     # backward_output_grad_names_map only contains the subset of module outputs that need a gradient,
                     # we filter out the invalid entries in grad_outputs, accessing using the mapped index.
-
-                    for _, i in self._onnx_graphs_info.backward_output_grad_names_map.items():
-                        grad_output = grad_outputs[i]
-                        if not grad_output.is_contiguous():
-                            grad_output = grad_output.contiguous()
+                    contiguous_grad_outputs = []
+                    for i in range(len(grad_outputs)):
+                        if i in self._onnx_graphs_info.backward_output_grad_names_map.values():
+                            grad_output = grad_outputs[i]
+                            if not grad_output.is_contiguous():
+                                grad_output = grad_output.contiguous()
+                            contiguous_grad_outputs.append(grad_output)
+                    # using contiguous_grad_outputs[] for holding grad_output to avoid data corruption issue in YieldOp
```

Comment:
The comment is not clear.
How collecting gradient output in to a python list hear is going to help data corruption in YieldOp?

Reply:
done

---
In reply to: [587854510](https://github.com/microsoft/onnxruntime/pull/6892#discussion_r587854510) [](ancestors = 587854510)